google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreOptions.java

```diff
@@ -216,8 +221,74 @@ public FirestoreOptions build() {
         }
       }
 
+      // Override credentials and channel provider if we are using the emulator.
+      String emulatorHost = System.getenv(FIRESTORE_EMULATOR_SYSTEM_PROPERTY);
+      if (emulatorHost != null) {
+        String hostUrlString = "http://" + emulatorHost;
+
+        // Try creating a host in order to validate that the host name is valid.
+        try {
+          URI.create(hostUrlString);
+        } catch (Exception e) {
+          throw new IllegalArgumentException(
+              "Value: '"
+                  + emulatorHost
+                  + "' for property "
+                  + FIRESTORE_EMULATOR_SYSTEM_PROPERTY
+                  + " is not a valid host",
+              e);
+        }
```

Comment:
can we extract this into a helper method `validateUrl`?

Reply:
We're not using this anywhere else, and the error code is pretty specific in that it's checking for the host.